src/Drivers/HubbardDriver.cpp

```diff
@@ -0,0 +1,59 @@
+#include "Drivers/HubbardDriver.hpp"
+
+namespace GQCP {
+
+    HubbardDriver::HubbardDriver(std::string csline, size_t num_states, size_t num_alpha, size_t num_beta, size_t K) {
+
+        std::string triagonal_line = csline;
+
+        // Read the upper triagonal of the hopping matrix
+        GQCP::VectorX<double> triagonal;
+        if (csline.empty()) {
+            throw std::invalid_argument("hubbard(driver): Comma-separated was empty!");
+        }
+
+        // Split comma-separated line
+        std::vector<std::string> splitted_line;
+        boost::split(splitted_line, triagonal_line, boost::is_any_of(","));
+
+        std::vector<double> triagonal_data;
+        for (const std::string &x : splitted_line) {
+            triagonal_data.push_back(std::stod(x));
+        }
+
+        triagonal = Eigen::Map<Eigen::VectorXd>(triagonal_data.data(), triagonal_data.size());
+        GQCP::HoppingMatrix H = GQCP::HoppingMatrix::FromUpperTriangle(triagonal);
```

Comment:
I think it would be useful if the previous lines of code that produces a `HoppingMatrix` from a `std::string` was actually made into a named constructor inside the `HoppingMatrix` class.